app/models/container_node.rb

```diff
@@ -75,6 +75,12 @@ def ipaddress
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddress, :port => 9090)
+    if MiqServer.active_miq_servers.where(:has_active_cockpit_ws => true)
```

Comment:
If possible, as discussed in gitter, it would be nice to not activate the role on servers without the required software installed so just checking for servers with the role active would be enough.  Let me think about it.